src/Controller.php

```diff
@@ -51,27 +51,34 @@ public function __construct(
      */
     public static function init()
     {
-        $opts = getopt('', array('dir:','replace'));
+        list($composer, $mode, $output) = self::getParams();
         return Container::create()
-            ->addValue(
-                'SplFileInfo.file_name',
-                dirname(__DIR__).DIRECTORY_SEPARATOR.'templates'
-            )
-            ->addValue(
-                'PhpParser\Parser.options',
-                array('throwOnError'=>true)
-            )
-            ->addValue(
-                'De\Idrinth\TestGenerator\Interfaces\JsonFile.file',
-                (isset($opts['dir']) && $opts['dir'] ?
-                        rtrim($opts['dir'], DIRECTORY_SEPARATOR) :
-                        getcwd()
-                    ).DIRECTORY_SEPARATOR.'composer.json'
-            )
-            ->addValue('De\Idrinth\TestGenerator\Interfaces\ClassWriter.replace', isset($opts['replace']))
+            ->addValue('SplFileInfo.file_name', dirname(__DIR__).DIRECTORY_SEPARATOR.'templates')
+            ->addValue('PhpParser\Parser.options', array('throwOnError' => true))
+            ->addValue('De\Idrinth\TestGenerator\Interfaces\JsonFile.file', $composer)
+            ->addValue('De\Idrinth\TestGenerator\Interfaces\ClassWriter.mode', $mode)
+            ->addValue('De\Idrinth\TestGenerator\Interfaces\Composer.output', $output)
             ->get(__CLASS__);
     }
 
+    /**
+     * processes getopt
+     * @return array [$composer, $mode, $output]
```

Comment:
is this a valid documentation? xD

Reply:
yes, @return type comment, see https://docs.phpdoc.org/references/phpdoc/tags/return.html